vagrant/lib/forklift/box_distributor.rb

```diff
@@ -60,13 +60,31 @@ def distribute!
       end
     end
 
+    def fetch_from_box_or_settings(box, key, default_value)
+      if !(value_from_box = box.fetch(key, nil)).nil?
+        value_from_box
+      else
+        @settings.fetch(key, default_value)
+      end
```

Comment:
The goal of [fetch](https://ruby-doc.org/core-3.1.2/Hash.html#method-i-fetch) is to provide a default if it isn't set. You can use that to simplify:
```suggestion
      box.fetch(key, @settings.fetch(key, default_value))
```

Or in a block if the computation of the value is expensive/dynamic:

```suggestion
      box.fetch(key) { @settings.fetch(key, default_value) }
```